crowbar_framework/app/controllers/application_controller.rb

```diff
@@ -37,6 +37,9 @@ class ApplicationController < ActionController::Base
     Rails.env.test? || \
     Rails.cache.fetch(:sanity_check_errors).empty?
   }
+  before_filter :upgrade, if: proc {
+    File.exist?("/var/run/crowbar/admin-server-upgrading")
```

Comment:
Well... this is only present when admin server is being upgraded. Don't we want to limit all non-upgrade calls during the while upgrade process?